File: 【哈希集合】2023Q1A-明明的随机数.cpp
/*
 * @hw id=2023Q1A lang=cpp
 *
 * 明明的随机数
 */

// @hw code=start
#include<iostream>
#include<vector>
#include<set>
#include<algorithm>
using namespace std;

int main() {
    int n;
    cin >> n;
    set<int> numset;
    for(int i=0;i<n;i++) {
        int num;
        cin >> num;
        numset.insert(num);
    }
    vector<int> numlist(numset.begin(), numset.end());
    sort(numlist.begin(), numlist.end());
    for(int num : numlist) {
        cout << num << endl;
    }
    return 0;
}
// @hw code=end

File: 【哈希集合】2023Q1A-明明的随机数.java
/*
 * @hw id=2023Q1A lang=java
 *
 * 明明的随机数
 */

// @hw code=start
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        Set<Integer> numset = new HashSet<>();
        for(int i=0;i<n;i++) {
            numset.add(in.nextInt());
        }
        List<Integer> numlist = new ArrayList<>(numset);
        Collections.sort(numlist);

        for(int num : numlist) {
            System.out.println(num);
        }
    }
}
// @hw code=end